### PR TITLE
modified changes for datetime and NaT Objects Behavior closes #43280

### LINKED ISSUE #43280: BUG: `pd.to_numeric` has an inconsistent behavior for `datetime` objects
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
>>> import pandas as pd
>>> from datetime import datetime
>>> pd.to_numeric(datetime(2021, 8, 22), errors="coerce")
nan
>>> pd.to_numeric(pd.Series(datetime(2021, 8, 22)), errors="coerce")
0    1629590400000000000
dtype: int64
>>> pd.Series([datetime(2021, 8, 22)]).apply(partial(pd.to_numeric), errors="coerce")
0   NaN
dtype: float64
>>>
>>> pd.to_numeric(pd.NaT, errors="coerce")
nan
>>> pd.to_numeric(pd.Series(pd.NaT), errors="coerce")
0   -9223372036854775808
dtype: int64
>>> pd.Series([pd.NaT]).apply(partial(pd.to_numeric), errors="coerce")
0   NaN
dtype: float64

```

#### Problem description

When using `pd.to_numeric` to convert a `pd.Series` with dtype `datetime64[ns]`, it returns different values than converting the series *value by value*

#### Expected Output

Converting a `pd.Series` as a *whole* should be the same than converting it *value by value*.
I am not sure about what the correct output should be, but IMO the output should be consistent in these two scenarios. 

What I suggest:
- For no-null values, returns the same value. Maybe the integer?
- For `pd.NaT`, always returns `np.NaN`

#### Output of ``pd.show_versions()``

I am using the latest version of `master` until today

<details>

INSTALLED VERSIONS
------------------
commit           : e39ea3024cebb4e7a7fd35972a44637de6c41650
python           : 3.8.3.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.6.0
Version          : Darwin Kernel Version 19.6.0: Tue Jun 22 19:49:55 PDT 2021; root:xnu-6153.141.35~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : en_US.UTF-8

pandas           : 1.4.0.dev0+517.gc3761e24d8
numpy            : 1.18.5
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 49.2.0.post20200714
Cython           : 0.29.21
pytest           : 5.4.3
hypothesis       : None
sphinx           : 3.1.2
blosc            : None
feather          : None
xlsxwriter       : 1.2.9
lxml.etree       : 4.5.2
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.16.1
pandas_datareader: None
bs4              : 4.9.1
bottleneck       : 1.3.2
fsspec           : 0.7.4
fastparquet      : None
gcsfs            : None
matplotlib       : 3.2.2
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.4
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.0
sqlalchemy       : 1.3.18
tables           : 3.6.1
tabulate         : 0.8.9
xarray           : None
xlrd             : 1.2.0
xlwt             : 1.3.0
numba            : 0.50.1

</details>

### PR DESCRIPTION
**Pull request by Navaneethan2503**
- [ #] Closes #43280
- [ #] tests added / passed

**Modified  datatime object 👍 ** 
i have changed view values to numeric of date and missing data found in pd series is returns Nan when coerce errors . 

**Example for solved changes 💯 **
```
>>> pd.to_numeric(pd.Series(datetime(2021, 8, 22)), errors="coerce")
[OUT]
0    20210822
dtype: int64

>>> pd.to_numeric(pd.Series(pd.NaT), errors="coerce")
[OUT]
0   NaN
dtype: float64

>>> s = pd.to_numeric(datetime(2021, 8, 22), errors="coerce")
0 20210822
dtype: Int64
>>> s = pd.to_numeric(pd.NaT, errors="coerce")
 nan

```

### CODE REVIEW

**Comment by hec10r** [comment]
Hi @Navaneethan2503, thank you for this PR. I think that this solves the problem for `pd.NaT`, but there is still an inconsistency for scalar arguments vs `pd.Series`:
```python
>>> import pandas as pd
>>> from datetime import datetime
>>> pd.to_numeric(pd.Series(datetime(2021,8,22)), errors="coerce")
0    20210822
dtype: int64
>>> pd.to_numeric(datetime(2021,8,22), errors="coerce")
nan
```
Additionally, I am not sure that the idea is to change the current behavior of datetime series for a new one. Instead, the idea should be to apply the current behavior to scalar arguments.

**Review by Kavyasilam — APPROVED** [review]
Good approach @Navaneethan2503

**Comment by Kavyasilam** [comment]
> Hi @Navaneethan2503, thank you for this PR. I think that this solves the problem for `pd.NaT`, but there is still an inconsistency for scalar arguments vs `pd.Series`:
> 
> ```python
> >>> import pandas as pd
> >>> from datetime import datetime
> >>> pd.to_numeric(pd.Series(datetime(2021,8,22)), errors="coerce")
> 0    20210822
> dtype: int64
> >>> pd.to_numeric(datetime(2021,8,22), errors="coerce")
> nan
> ```
> 
> Additionally, I am not sure that the idea is to change the current behavior of datetime series for a new one. Instead, the idea should be to apply the current behavior to scalar arguments.

I thing all scalar argument are passed , my suggestion idea is good because datatime behaviour are all type passed

**Comment by jreback** [comment]
@Navaneethan2503 pls do not open multiple PRs for the same issue
close one now

and just push to the same one

**Review by jreback — CHANGES_REQUESTED** [review]
always always start with tests that fail w/o the change and pass with

**Comment by Navaneethan2503** [comment]
Final PR pass all scalar arguments including datetime and NaT DataType , make it merge

File: pandas/core/tools/numeric.py [hunk 1]
```diff
@@ -130,6 +130,19 @@ def to_numeric(arg, errors="raise", downcast=None):
...
+    >>> s = pd.to_numeric(datetime(2021, 8, 22), errors="coerce")
+    0 20210822
+    dtype: Int64
+    >>> s = pd.to_numeric(pd.NaT, errors="coerce")
+    nan
+    >>> s = pd.to_numeric(pd.Series(datetime(2021, 8, 22)), errors="coerce")
+    0 20210822
+    dtype: Int64
+    >>> s = pd.to_numeric(pd.Series(pd.NaT), errors="coerce")
+    NaN
```
Comment (simonjayhawkins) [inline pandas/core/tools/numeric.py:145]: hmm. does the doctest pass on this? the output of an assignment should be None and nothing displayed?

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
I think the expected behavior is still being discussed? This is breaking tests.

File: pandas/core/tools/numeric.py (hunk 1 above)
Comment (Navaneethan2503) [reply pandas/core/tools/numeric.py:145]: Yes , Nothing will display

Comment (simonjayhawkins) [reply pandas/core/tools/numeric.py:145]: please let the question asker mark as resolved

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
you have requested review but I see no changes since the previous review and ci is failing?

also can you move the tests from the docstring to pandas/tests and add a release note

File: pandas/core/tools/numeric.py [hunk 2]
```diff
@@ -175,7 +180,7 @@ def to_numeric(arg, errors="raise", downcast=None):
     if is_numeric_dtype(values_dtype):
         pass
     elif is_datetime_or_timedelta_dtype(values_dtype):
-        values = values.view(np.int64)
+
```
Comment (simonjayhawkins) [inline pandas/core/tools/numeric.py:183]: ci failing. `IndentationError: expected an indented block`

**Comment by mroeschke** [comment]
Thanks for the PR but it appears this is a duplicate of https://github.com/pandas-dev/pandas/pull/43315 which is a little further along. Closing as a duplicate but happy to have you work on other issues with a `good first issue` label.

**Pull request closed without merging**